includes/wikia/WikiaDataAccess.class.php

```diff
@@ -302,6 +302,18 @@ static private function getDataAndVerify( $app, $key, $clearLocalCache = false )
 		return $result;
 	}
 
+	static public function criticalSection( $key, $lockTimeout, callable $fn, &$result = null ) {
```

Comment:
Why do we need to return value by reference in this method?
Maybe we could throw an exception when lock cannot be acquired?